### PR TITLE
let `datagen` support pzipfx dataset

### PR DESCRIPTION
**Pull request by rebelice**
Usage:
optimizer-tester datagen --dataset=pzipfx --dir={dir} --args={args}

pzipfx:
 实例：`--args="x=1.2,n=1000000,ndv=1000,partition_num=4"`
 
args 参数 | 说明 |
-- | -- |
x | zipf 的分布参数，需要大于 1 |
n | 生成的行数
ndv | NDV
partition_num| 分区数量

### CODE REVIEW

**Comment by rebelice** [comment]
@qw4990 @Reminiscent PTAL

File: datagen/util.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+const timeLayout = "2006-01-02 15:04:05"
+
+func prepareIntNDV(ndv int) []int {
+	ints := make([]int, ndv)
+	intMap := make(map[int]bool)
+	for i := range ints {
+		var ok bool
+		var data int
+		for {
+			data = rand.Intn(int(2e9))
```
Comment (Reminiscent) [inline datagen/util.go:17]: Add a check for NDV to prevent this endless loop.

Comment (Reminiscent) [reply datagen/util.go:17]: In addition, will there be any problems when the NDV is the same?

File: datagen/dataset_pzipfx.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+	content := fmt.Sprintf("CREATE TABLE %s (a %s, b %s, c %s, key(a), key(a, b), key(b), key(b, c))\n",
+		t.name,
+		t.dataType,
+		t.dataType,
+		t.dataType)
+
+	switch t.parType {
+	case HashPart:
+		content = fmt.Sprintf("%s PARTITION BY HASH(%s) PARTITION %d;", content, t.parCol, opt.partitionNum)
```
Comment (Reminiscent) [inline datagen/dataset_pzipfx.go:143]: Maybe we can add some checks for the column's type.

File: datagen/dataset_pzipfx.go [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+		begin, _ := time.Parse(timeLayout, "2000-01-01 00:00:00")
+		end, _ := time.Parse(timeLayout, "2100-01-01 00:00:00")
+		interval := end.Sub(begin)
+
+		var i int64
+		for i = 0; i < opt.partitionNum-1; i++ {
+			var v string
+			switch t.dataType {
+			case TypeInt:
+				v = strconv.FormatUint(uint64(ints[int64(len(ints))*i/opt.partitionNum]), 10)
```
Comment (Reminiscent) [inline datagen/dataset_pzipfx.go:156]: Please use a more random method to generate the partition range. For example, if the data range belongs to [0, 10000]. We can use 10000/partitionNum to generate the partition range.

Comment (qw4990) [reply datagen/dataset_pzipfx.go:156]: I'd like to merge this PR and solve this problem later.

**Pull request merged**